pkg/recipes/terraform/install.go

```diff
@@ -78,5 +81,42 @@ func Install(ctx context.Context, installer *install.Installer, tfDir string) (s
 
 	logger.Info(fmt.Sprintf("Terraform latest version installed to: %q", execPath))
 
-	return execPath, nil
+	// Create a new instance of tfexec.Terraform with current Terraform installation path
+	tf, err := NewTerraform(ctx, tfDir, execPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Verify Terraform installation is complete before proceeding
+	for attempt := 0; attempt <= installVerificationRetryCount; attempt++ {
+		_, _, err = tf.Version(ctx, false)
+		if err == nil {
+			metrics.DefaultRecipeEngineMetrics.RecordTerraformInstallVerificationDuration(ctx, installStartTime,
+				[]attribute.KeyValue{
+					metrics.TerraformVersionAttrKey.String("latest"),
+					metrics.OperationStateAttrKey.String(metrics.SuccessfulOperationState),
+				},
+			)
+			break
+		}
+		if err != nil {
```

Comment:
We don't need this. If `tf.Version` is successful, then it is going to go into the `if err == nil`, and then the break will exit the for loop. Each time `tf.Version` returns an error, it will go to the second if. So, we don't need the second if I think.

tf.Version
=> Success => Exit for loop
=> Failure => Sleep and retry

Reply:
Good catch, this originally was only checking `err != nil` but I missed removing that check once `err == nil` was added